download.go

```diff
@@ -171,12 +174,55 @@ func readAndCheckImage(r io.Reader, contentLength int) (*imageData, error) {
 	return &imageData{buf.Bytes(), imgtype, cancel}, nil
 }
 
-func requestImage(imageURL string) (*http.Response, error) {
+func requestImage(imageURL string, r *http.Request) (*http.Response, error) {
 	req, err := http.NewRequest("GET", imageURL, nil)
 	if err != nil {
 		return nil, newError(404, err.Error(), msgSourceImageIsUnreachable).SetUnexpected(conf.ReportDownloadingErrors)
 	}
 
+	if conf.CookiePassthrough && r != nil {
+		jar, err := cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
+		if err != nil {
+			return nil, err
+		}
+
+		cookieBase := conf.CookieBaseURL
+
+		if len(cookieBase) == 0 {
+			scheme := r.Header.Get("X-Forwarded-Proto")
+			if len(scheme) == 0 {
+				scheme = "https"
```

Comment:
Since imgproxy doesn't support HTTPS, it seems that the default scheme should be `http`